exporters/etw/include/opentelemetry/exporters/etw/etw_provider_exporter.h

```diff
@@ -0,0 +1,658 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#ifndef WIN32_LEAN_AND_MEAN
+#  define WIN32_LEAN_AND_MEAN
+#endif
+
+#ifdef _MSC_VER
+// evntprov.h(838) : warning C4459 : declaration of 'Version' hides global declaration
+#  pragma warning(disable : 4459)
+// needed for Unit Testing with krabs.hpp
+#  pragma warning(disable : 4018)
+#endif
+
+#include "opentelemetry/common/attribute_value.h"
+#include "opentelemetry/exporters/etw/uuid.h"
+#include "opentelemetry/version.h"
+
+#include "opentelemetry/exporters/etw/utils.h"
+
+#ifdef HAVE_MSGPACK
+// This option requires INCLUDE_DIR=$(ProjectDir)\..\..\third_party\json\include;...
+#  include "nlohmann/json.hpp"
+#endif
+
+#ifndef HAVE_NO_TLD
+// Allow to opt-out from `TraceLoggingDynamic.h` header usage
+#  define HAVE_TLD
+#  include "TraceLoggingDynamic.h"
+#endif
+
+#include <map>
+#include <mutex>
+#include <unordered_map>
+#include <vector>
+
+#ifdef HAVE_KRABS_TESTS
+// krabs.hpp requires this definition of min macro from Windows.h
+#  ifndef min
+#    define min(a, b) (((a) < (b)) ? (a) : (b))
+#  endif
+#endif
+
+#define MICROSOFT_EVENTTAG_NORMAL_PERSISTENCE 0x01000000
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+
+class ETWProvider
+{
+
+public:
+  const unsigned long STATUS_OK    = 0;
+  const unsigned long STATUS_ERROR = ULONG_MAX;
+  const unsigned long STATUS_EFBIG = ULONG_MAX - 1;
+
+  enum EventFormat
+  {
+    ETW_MANIFEST = 0,
+    ETW_MSGPACK  = 1,
+    ETW_XML      = 2
+  };
+
+  /// <summary>
+  /// Entry that contains Provider Handle, Provider MetaData and Provider GUID
+  /// </summary>
+  struct Handle
+  {
+    REGHANDLE providerHandle;
+    std::vector<BYTE> providerMetaVector;
+    GUID providerGuid;
+  };
+
+  /// <summary>
+  /// Check if given provider is registered.
+  /// </summary>
+  /// <param name="providerId"></param>
+  /// <returns></returns>
+  bool is_registered(const std::string &providerId)
```

Comment:
Name functions and methods according to [they style guide](https://google.github.io/styleguide/cppguide.html#Function_Names), here and in all other places.